k8s/operators/src/pool/diskpool/client.rs

```diff
@@ -178,6 +178,41 @@ pub(crate) async fn migrate_to_v1beta1(
     Ok(())
 }
 
+/// List pool from control plane. Checks if DSP cr present for it. If no, create it ourself.
+pub(crate) async fn create_missing_cr(
+    k8s: &Client,
+    control_client: clients::tower::ApiClient,
+    namespace: &str,
+) -> Result<(), Error> {
+    let pools = control_client.pools_api().get_pools().await.unwrap();
+    let pools_api: Api<DiskPool> = v1beta1_api(k8s, namespace);
+    let param = PostParams::default();
+    for pool in pools.into_body().iter_mut() {
+        match pools_api.get(&pool.id).await {
+            Err(kube::Error::Api(e)) if e.code == StatusCode::NOT_FOUND => {
+                info!(pool.id, "Found pool to be missing in k8s");
+                if let Some(spec) = pool.spec.clone() {
+                    let spec: DiskPoolSpec = DiskPoolSpec::new(spec.node, spec.disks);
+                    let new_disk_pool: DiskPool = DiskPool::new(&pool.id, spec);
```

Comment:
Should we add any labels here to reflect that it was automatically created?
CC @Abhinandan-Purkait

Reply:
@tiagolobocastro  As discussed, We will add this later when we will have operations around labels in CR.